src/ir/item.rs

```diff
@@ -346,6 +366,10 @@ pub struct Item {
     parent_id: ItemId,
     /// The item kind.
     kind: ItemKind,
+    /// Detect cycles when determining if we can derive debug/copy or not, and
+    /// avoid infinite recursion.
+    detect_derive_debug_cycle: Cell<bool>,
```

Comment:
Can we remove the relevant fields from `CompInfo`? If not, why?

Is it worth to add an `struct AutoRestore<T: Copy>(&'a Cell<T>)` to simplify the code flow here? If so, mind filing an issue for it?